src/node_file.cc

```diff
@@ -1594,6 +1600,45 @@ static void RMDir(const FunctionCallbackInfo<Value>& args) {
   }
 }
 
+static void RimrafUnlinkSync(const FunctionCallbackInfo<Value>& args) {
+  const int argc = args.Length();
+  CHECK_EQ(argc, 3);  // path, retries, retrydelay
+
+  CHECK(args[1]->IsInt32());
+  CHECK(args[2]->IsInt32());
+
+  Environment* env = Environment::GetCurrent(args);
+  BufferValue path(env->isolate(), args[0]);
+
+  THROW_IF_INSUFFICIENT_PERMISSIONS(
+      env, permission::PermissionScope::kFileSystemWrite, path.ToStringView());
+
+  const auto tries = args[1].As<Int32>()->Value() + 1;
+  const auto retry_delay = args[2].As<Int32>()->Value();
+  constexpr std::array<int, 5> retryable_errors = {
+      EBUSY, EMFILE, ENFILE, ENOTEMPTY, EPERM};
+
+  uv_fs_t req;
+
+  for (int i = 1; i <= tries; i++) {
+    FS_SYNC_TRACE_BEGIN(unlink);
+    auto err = uv_fs_unlink(nullptr, &req, *path, nullptr);
```

Comment:
If it doesn't throw you need to break the loop. You forgot the return statement that JS loop executes. If first iteration is successful, you shouldn't try again